pkg/command/experiments.go

```diff
@@ -0,0 +1,182 @@
+			q.SetLabelSelector(parseLabelSelector(selector))
+			if err := l.ForEachExperiment(ctx, q, result.Add); err != nil {
+				return err
+			}
+		}
+
+		return p.Fprint(out, result)
+	}
+
+	cmd.Flags().IntVar(&batchSize, "batch-size", batchSize, "fetch large lists in chu`n`ks rather then all at once")
+	cmd.Flags().StringVarP(&selector, "selector", "l", selector, "selector (label `query`) to filter on")
+
+	return cmd
+}
+
+// NewDeleteExperimentsCommand returns a command for deleting experiments.
+func NewDeleteExperimentsCommand(cfg Config, p Printer) *cobra.Command {
+	var (
+		ignoreNotFound bool
+	)
+
+	cmd := newExperimentsCommand(cfg)
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		ctx, out := cmd.Context(), cmd.OutOrStdout()
+		client, err := api.NewClient(cfg.Address(), nil)
+		if err != nil {
+			return err
+		}
+
+		l := experiments.Lister{
+			API: experiments.NewAPI(client),
+		}
+
+		return l.ForEachNamedExperiment(ctx, args, ignoreNotFound, func(item *experiments.ExperimentItem) error {
+			selfURL := item.Link(api.RelationSelf)
+			if selfURL == "" {
+				// TODO Should this fail?
+				return nil
+			}
+
+			if err := l.API.DeleteExperiment(ctx, selfURL); err != nil {
+				return err
+			}
+
+			return p.Fprint(out, item)
+		})
+	}
+
+	cmd.Flags().BoolVar(&ignoreNotFound, "ignore-not-found", ignoreNotFound, "treat not found errors as successful deletes")
+
+	return cmd
+}
+
+// NewLabelExperimentsCommand returns a command for labeling experiments.
+func NewLabelExperimentsCommand(cfg Config, p Printer) *cobra.Command {
+	cmd := newExperimentsCommand(cfg)
+	cmd.Use += " KEY=VAL ..."
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		ctx, out := cmd.Context(), cmd.OutOrStdout()
+		client, err := api.NewClient(cfg.Address(), nil)
+		if err != nil {
+			return err
+		}
+
+		l := experiments.Lister{
+			API: experiments.NewAPI(client),
+		}
+
+		names, labels := argsToNamesAndLabels(args)
+		return l.ForEachNamedExperiment(ctx, names, false, func(item *experiments.ExperimentItem) error {
+			labelsURL := item.Link(api.RelationLabels)
+			if labelsURL == "" {
+				// TODO Should this fail?
+				return nil
+			}
+
+			if err := l.API.LabelExperiment(ctx, labelsURL, experiments.ExperimentLabels{Labels: labels}); err != nil {
+				return err
+			}
+
+			return p.Fprint(out, item)
+		})
+	}
+
+	return cmd
+}
+
+// validExperimentArgs returns shell completion logic for experiment/trial names.
+func validExperimentArgs(cfg Config, singleMatchSuffix string) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		ctx := cmd.Context()
+		client, err := api.NewClient(cfg.Address(), nil)
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+
+		l := experiments.Lister{
+			API: experiments.NewAPI(client),
+		}
+
+		names := make([]string, 0, 16)
```

Comment:
any reason for 16 specifically?

Reply:
I don't even remember writing this. I have no idea.